vms/platformvm/state/state.go

```diff
@@ -1944,3 +2010,50 @@ func (s *state) writeMetadata() error {
 	}
 	return nil
 }
+
+// TODO: Remove after next hard-fork when everybody has run this.
+func (s *state) populateBlockHeightIndex() error {
+	blk, _, err := s.GetStatelessBlock(s.lastAccepted)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve last accepted block from local disk: %w", err)
+	}
+
+	heightKey := database.PackUInt64(blk.Height())
+	if _, err := database.GetID(s.blockIDDB, heightKey); err == nil {
+		// If the last accepted block is in [s.blockIDDB], no backfilling is required.
+		return nil
+	}
+
+	blockIterator := s.blockDB.NewIterator()
+	defer blockIterator.Release()
+	for blockIterator.Next() {
+		blkBytes := blockIterator.Value()
+
+		// Note: stored blocks are verified, so it's safe to unmarshal them with GenesisCodec
+		blkState := stateBlk{}
+		if _, err := blocks.GenesisCodec.Unmarshal(blkBytes, &blkState); err != nil {
+			return err
+		}
+
+		blkState.Blk, err = blocks.Parse(blocks.GenesisCodec, blkState.Bytes)
+		if err != nil {
+			return err
+		}
+
+		blkHeight := blkState.Blk.Height()
+		blkID := blkState.Blk.ID()
+
+		// Populate the in-memory cache with the [blockIDCacheSize] most recent blocks
+		if math.AbsDiff(s.currentHeight, blkHeight) <= blockIDCacheSize {
+			s.addedBlockIDs[blkHeight] = blkID
+			s.blockIDCache.Put(blkHeight, blkID)
+		}
```

Comment:
Should we do this? cc @StephenButtolph 

Pre-populating the cache here _feels_ good since we're iterating through the entire block db anyway but not sure if there's an actual benefit.

Reply:
question: instead of iterating blocks by database native iterator, should we instead hop from the last block to its parent and its ancestors in order?
Not sure if performances wise there is a difference, but the cache population would be more natural

Reply:
I think the performance would be similar. The cache population would be more natural but I'm not sure if we're going forward with the cache population